pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -250,3 +288,61 @@ func newTestHTTPClient(response *http.Response, err error) *http.Client {
 		},
 	}
 }
+
+type benchmarkingRoundTripper struct {
+	dataType benchmarkingDataType
```

Comment:
I needed to make the response dynamic because otherwise resp.body will be consumed and it will always returns zero, so I cannot just use the fakeRoundtripper. Verified that with the following test.
```
func TestHTTPSBB(t *testing.T) {
	hClient := newBenchmarkingtHTTPClient()
	scrapeClient, err := newHTTPScrapeClient(hClient)
	if err != nil {
		t.Fatal(err)
	}
	for i := 0; i < 3; i++ {
		if _, err := scrapeClient.Scrape(testURL); err != nil {
			t.Fatal(err)
		}
	}
}
=== RUN   TestHTTPSBB
{a-lzrjc-deployment-85d4b7d859-gspjs 2.3 100.2 0 100 12.2}
{ 0 0 0 0 0}
{ 0 0 0 0 0}
--- PASS: TestHTTPSBB (0.00s)
PASS

```